### PR TITLE
Add citation file.

### PR DESCRIPTION
**Pull request by zoq**
Add citation file to use GitHub's citation feature.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    affiliation: RelationalAI
+    family-names: Curtin
+    given-names: "Ryan R."
+  -
+    affiliation: "Free University of Berlin"
+    family-names: Edel
+    given-names: Marcus
+  -
+    affiliation: "Birla Institute of Technology and Science"
+    family-names: "Prabhu "
```
Comment (rcurtin) [inline CITATION.cff:15]: Seems like no need for an extra space here?

**Comment by zoq** [comment]
Hm, not sure why the mlpack integration test fails, maybe this was just a fluke.

**Comment by zoq** [comment]
@mlpack-jenkins test this please

**Review by rcurtin — APPROVED** [review]
Looks good to me---excited to see how this looks once it's merged. :)

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  -
+    affiliation: "University of Texas"
+    family-names: Basak
+    given-names: Suryoday
+  -
+    affiliation: Epsilon
+    family-names: Lou
+    given-names: Zhihao
+  -
+    affiliation: "Data61/CSIRO, Griffith University"
```
Comment (conradsnicta) [inline CITATION.cff:26]: @zoq should be "Data61/CSIRO and Griffith University"  
(it's two separate organisations)
```suggestion
    affiliation: "Data61/CSIRO and Griffith University"
```

**Review by conradsnicta — CHANGES_REQUESTED** [review]

**Review by conradsnicta — APPROVED** [review]

**Comment by conradsnicta** [comment]
@zoq @rcurtin so after merging this, turns out GitHub prefers to provide the wrong citation when clicking on "Cite this repository" (left hand side on desktop view).

Instead of citing the JMLR article, this is what's provided by GitHub:
```@misc{Curtin_The_ensmallen_library,author = {Curtin, Ryan R. and Edel, Marcus and Prabhu, Rahul Ganesh and Basak, Suryoday and Lou, Zhihao and Sanderson, Conrad},title = {{The ensmallen library for flexible numerical optimization}},url {https://github.com/mlpack/ensmallen}}```

**Pull request merged**